ml_metadata/tools/mlmd_bench/stats.cc

```diff
@@ -65,22 +67,40 @@ void ThreadStats::Merge(const ThreadStats& other) {
   finish_ = std::max(finish_, other.finish());
 }
 
-void ThreadStats::Report(const std::string& specification) {
-  std::string extra;
+void ThreadStats::Report(const std::string& specification,
+                         WorkloadConfigResult* workload_summary) {
+  if (done_ == 0) {
+    LOG(ERROR) << "Current workload has not been executed even once!";
+    return;
+  }
+
+  double bytes_per_second = 0.0;
+  double microseconds_per_operation =
+      (accumulated_elapsed_time_ / absl::Microseconds(1)) / done_;
+
+  std::string rate;
+  // Not all workloads will transfer bytes in the process.
+  // Also, if `elapsed_seconds` equals to 0, `bytes_per_second` will remain as
+  // 0.
   if (bytes_ > 0) {
+    // Refines the granularity of time to microsecond first and convert it back
+    // to second.
+    double elapsed_seconds = (finish_ - start_) / absl::Microseconds(1) * 1e-6;
     // Rate is computed on actual elapsed time (latest end time minus
     // earliest start time of each thread) instead of the sum of per-thread
     // elapsed times.
-    int64 elapsed_seconds = accumulated_elapsed_time_ / absl::Seconds(1);
-    std::string rate =
-        absl::StrFormat("%6.1f KB/s", (bytes_ / 1024.0) / elapsed_seconds);
-    extra = rate;
+    bytes_per_second = bytes_ / elapsed_seconds;
```

Comment:
do you need to handle elapsed_seconds == 0? as stated in the comment?

Reply:
Discussed above.